### DISCUSSION TITLE
How to add parameters to heifsave svt encoder?

### DISCUSSION THREAD

**Discussion by ty114514** [libvips/libvips#4309] [Q&A]
I want to use https://gitlab.com/AOMediaCodec/SVT-AV1/-/blob/master/Docs/Parameters.md, but `vips` doesn't seem to have something like `ffmpeg -svtav1-params`?

**Comment by jcupitt** [comment]
Hello, here's the libheif saver, you'd need to add some code to that. Assuming libheif exposes the param you want to set.

https://github.com/libvips/libvips/blob/master/libvips/foreign/heifsave.c

**Reply by ty114514** [reply]
Thanks for your reply. I suddenly realized that it is indirectly called through `libheif` instead of directly accessing `libsvtav1`.
Sorry I didn't explain my purpose. I am processing some ultra-high resolution images. I need to set the preset of `libsvtav1` to be greater than 8, but vips does not support it and will prompt
```
Svt[error]: Instance 1: 8k+ resolution support is limited to M8
```
My coding skills are not very high, and after a quick look at `libheif` it seems that it does not provide these parameters. So can I only use ffmpeg? Do you have any suggestions for this? Thanks again!

**Reply by jcupitt** [reply]
Ah that's a shame. You could ask libheif to add support, they are usually very responsive.

Otherwise, yes, I suppose it's ffmpeg.
